src/ord/map.rs

```diff
@@ -314,17 +285,12 @@ where
     /// another, i.e. the set of entries to add, update, or remove to
     /// this map in order to make it equal to the other map.
     ///
-    /// This function will avoid visiting nodes which are shared
-    /// between the two maps, meaning that even very large maps can be
-    /// compared quickly if most of their structure is shared.
-    ///
-    /// Time: O(n) (where n is the number of unique elements across
-    /// the two maps, minus the number of elements belonging to nodes
-    /// shared between them)
```

Comment:
Is there some reason we can't continue to support the efficient diffs? It's mostly based on pointer equality comparisons, so I don't see why B-tree vs B+tree would make a difference...

(Doesn't have to be in this PR -- I can take a look later. Just trying to understand the implications)

Reply:
This should be fully doable. I was just pacing myself so I could get the main body of the work done first.

Reply:
I restored this optimization partially